extensions/plugin-history-sync/src/historySyncPlugin.tsx

```diff
@@ -142,12 +142,17 @@ export function historySyncPlugin<
           (r) => r.activityName === fallbackActivityName,
         );
         const fallbackActivityPath = fallbackActivityRoute?.path;
+        const fallbackActivityParams = Object.fromEntries(
+          pathToUrl(currentPath).searchParams.entries(),
+        );
```

Comment:
```suggestion
        const fallbackActivityParams = urlSearchParamsToMap(
          pathToUrl(currentPath).searchParams,
        );
```

`urlSearchParamsToMap()` seems to accomplish a similar task to what you've implemented here.

To improve code reuse and maintainability, could you consider refactoring this section to utilize it instead of duplicating the logic?

This will help keep the codebase DRY and make future updates easier.

Reply:
resolved